File: Modules/ServerTypes/PerformanceAppraisal/PerformanceAppraisalEvaluationForm.ts
import { TextAreaEditor, DecimalEditor, ImageUploadEditor, PrefixedContext } from "@serenity-is/corelib";
import { initFormType } from "@serenity-is/corelib/q";

export interface PerformanceAppraisalEvaluationForm {
    Evaluation: TextAreaEditor;
    Goals: TextAreaEditor;
    Summary: TextAreaEditor;
    BonusRate: DecimalEditor;
    OverallRate: DecimalEditor;
    EmployeeSignature: ImageUploadEditor;
    HodSignature: ImageUploadEditor;
    GeneralManagerSignature: ImageUploadEditor;
}

export class PerformanceAppraisalEvaluationForm extends PrefixedContext {
    static formKey = 'PerformanceAppraisal.PerformanceAppraisalEvaluation';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!PerformanceAppraisalEvaluationForm.init)  {
            PerformanceAppraisalEvaluationForm.init = true;

            var w0 = TextAreaEditor;
            var w1 = DecimalEditor;
            var w2 = ImageUploadEditor;

            initFormType(PerformanceAppraisalEvaluationForm, [
                'Evaluation', w0,
                'Goals', w0,
                'Summary', w0,
                'BonusRate', w1,
                'OverallRate', w1,
                'EmployeeSignature', w2,
                'HodSignature', w2,
                'GeneralManagerSignature', w2
            ]);
        }
    }
}

File: Modules/HumanResource/PerformanceAppraisal/PerformanceAppraisalEvaluation/PerformanceAppraisalEvaluationForm.cs
using Serenity.ComponentModel;
using System;

namespace HRMSoftware.PerformanceAppraisal.Forms;

[FormScript("PerformanceAppraisal.PerformanceAppraisalEvaluation")]
[BasedOnRow(typeof(PerformanceAppraisalEvaluationRow), CheckNames = true)]
public class PerformanceAppraisalEvaluationForm
{
    [TextAreaEditor(Cols = 3, Rows = 4)]
    public string Evaluation { get; set; }
    
    [TextAreaEditor(Cols = 3, Rows = 4), HalfWidth]
    public string Goals { get; set; }
    
    [TextAreaEditor(Cols = 3, Rows = 4), HalfWidth]
    public string Summary { get; set; }
    
    [HalfWidth]
    public decimal BonusRate { get; set; }
    
    [HalfWidth]
    public decimal OverallRate { get; set; }
    
    [HalfWidth]
    public string EmployeeSignature { get; set; }
    [HalfWidth]
    public string HodSignature { get; set; }
    [HalfWidth, AlignCenter]
    public string GeneralManagerSignature { get; set; }
}